### PR TITLE
Updated Testnet release notes and added InterSphinx

### PR DESCRIPTION
**Pull request by jens-concordium**
## Purpose

The Testnet release notes were out-dated, so these have been updated to match the current releases.

## Changes

Testnet release notes were updated, and InterSphinx was added to enable references between Mainnet and Testnet.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

File: source/testnet/conf.py [hunk 1]
```diff
@@ -260,3 +261,7 @@
         "Accept-Encoding": "utf-8",
     }
 }
+
+
+# -- InterSphinx links for linking between document versions ------------------
+intersphinx_mapping = {'mainnet': ('http://developer.concordium.software/en/mainnet/', None)}
```
Comment (limemloh) [inline source/testnet/conf.py:267]:
```suggestion
intersphinx_mapping = {'mainnet': ('https://developer.concordium.software/en/mainnet/', None)}
```

File: source/mainnet/conf.py [hunk 2]
```diff
@@ -260,3 +261,6 @@
         "Accept-Encoding": "utf-8",
     }
 }
+
+# -- InterSphinx links for linking between document versions ------------------
+intersphinx_mapping = {'testnet': ('http://developer.concordium.software/en/testnet/', None)}
```
Comment (limemloh) [inline source/mainnet/conf.py:266]:
```suggestion
intersphinx_mapping = {'testnet': ('https://developer.concordium.software/en/testnet/', None)}
```

Comment (limemloh) [reply source/mainnet/conf.py:266]: Should probably be https just for good measure

File: source/testnet/net/resources/release-notes.rst [hunk 3]
```diff
@@ -8,6 +8,103 @@ Release Notes
...
+Concordium blockchain has native support for smart contracts on-chain with our core on-chain language WebAssembly (Wasm), a portable well-defined assembly-like language.
+
+Rust is the first off-chain high level smart contract language.
+
+
+**Tokenomics and On-chain Incentivization**
+
+The Concordium blockchain comprises a set of transactions and economic roles that interact within the economy. An economic role, such as a baker or account holder, is represented by an account on the Concordium platform.
+
+The flow of GTU between accounts via transactions creates an economy that is designed to incentivize participation in the network and counter dishonest behaviour. It is the objective of the Concordium Foundation to guide the creation of a sustainable economy that rewards participants for their efforts in developing the network.
```
Comment (LotteKh) [inline source/testnet/net/resources/release-notes.rst:95]:
```suggestion
The flow of GTU between accounts via transactions creates an economy that is designed to incentivize participation in the network and counter dishonest behavior.
```

File: source/testnet/net/resources/release-notes.rst [hunk 4]
```diff
@@ -8,6 +8,103 @@ Release Notes
...
+
+Open Testnet v5 Update 2
+========================
+
+**Concordium Desktop Wallet v1.1.3 for Testnet.**
+
+The Desktop Wallet is available on Testnet for Windows, macOS, and Linux including:
+
+* All features released in v1.0.2 for Mainnet.
+* Transaction status is now available in account reports.
```
Comment (LotteKh) [inline source/testnet/net/resources/release-notes.rst:22]:
```suggestion
* Transaction status in account reports.
```

File: source/testnet/net/resources/release-notes.rst [hunk 5]
```diff
@@ -8,6 +8,103 @@ Release Notes
...
+* Various bug fixes.
+* Foundation feature: Added support for bulk import of proposals
```
Comment (LotteKh) [inline source/testnet/net/resources/release-notes.rst:24]:
```suggestion
* Foundation feature: Added support for bulk import of proposals.
```

File: source/testnet/net/resources/release-notes.rst [hunk 6]
```diff
@@ -8,6 +8,103 @@ Release Notes
...
+* Code is now open source.
+
+.. _open-testnet-v5:
+
+Open Testnet v5
+===============
+
+May 12th, 2021
+
+Updated Open Testnet to match Mainnet features, which includes:
```
Comment (LotteKh) [inline source/testnet/net/resources/release-notes.rst:62]:
```suggestion
Updated Open Testnet to match Mainnet features including:
```

File: source/testnet/net/resources/release-notes.rst [hunk 7]
```diff
@@ -8,6 +8,103 @@ Release Notes
...
+
+
+**Proof of Stake**
+
+The Concordium Blockchain uses a proof of stake mechanism to ensure resource-efficient operation of the network.
+
+
+**Two Layer Consensus Protocol**
+
+Nakamoto-Style Consensus Bakers participate in a form of lottery to win the right to append blocks to the chain.
```
Comment (LotteKh) [inline source/testnet/net/resources/release-notes.rst:72]: I think there's a missing colon (and it's also missing from the mainnet docs, so it should be changed there as well.)
Nakamoto-Style Consensus:  Bakers participate in

File: source/testnet/net/resources/release-notes.rst [hunk 8]
```diff
@@ -8,6 +8,103 @@ Release Notes
...
+
+Finality Layer Concordium finality layer dynamically ‘checkpoints’ the blockchain using Byzantine agreement to identify and mark common blocks in the chains of honest users as final.
```
Comment (LotteKh) [inline source/testnet/net/resources/release-notes.rst:74]: Same as above - missing colon.
Finality Layer: The Concordium finality layer ...

File: source/testnet/net/resources/release-notes.rst [hunk 9]
```diff
@@ -8,6 +8,103 @@ Release Notes
...
+**Built in IDLayer**
+
+Account creation is based on a validated identity, but at the same time it provides transactional privacy for users with a mechanism that allows accountability to local regulatory authorities.
+
+Transactional privacy is further enhanced by support for shielded transfers.
+
+
+**Smart Contracts**
+
+Concordium blockchain has native support for smart contracts on-chain with our core on-chain language WebAssembly (Wasm), a portable well-defined assembly-like language.
```
Comment (LotteKh) [inline source/testnet/net/resources/release-notes.rst:86]:
```suggestion
Concordium blockchain has native support for smart contracts on-chain with our core on-chain language WebAssembly (Wasm), a portable, well-defined, assembly-like language.
```

**Review by LotteKh — APPROVED** [review]

**Pull request merged**